platforms/core-configuration/bean-serialization-services/build.gradle.kts

```diff
@@ -26,13 +26,15 @@ dependencies {
     api(projects.stdlibJavaExtensions)
     api(projects.modelCore)
     api(projects.persistentCache)
+    api(projects.serviceProvider)
```

Comment:
Not saying it shouldn't be, but could you share an example of code in this module that requires `serviceProvider` to be an `api` dependency?

Reply:
`BeanSerializationServices` extends `AbstractGradleModuleServices`

Reply:
Could you expand on why that requires us to include the `serviceProvider` module in the build path of consumers of `bean-serialization-services`? Is the type `BeanSerializationServices` supposed to be consumed by clients of this module, for instance?

I was under the (possibly wrong) impression that as long as a type is not meant to be exposed to consuming modules, then all dependencies it requires could be `implementation`.

Reply:
While consumers aren't supposed to interact directly with the type, they still need to instantiate it via the SPI interface which requires the type to be public. And since the type is public, `projectHealth` considers the type and its supertypes part of the public interface, thus `api`.